docs/topics/periodic-stats.rst

```diff
@@ -0,0 +1,298 @@
+.. _topics-periodic-stats:
+
+==============
+Periodic Stats
+==============
+
+Sometimes you want to monitor statistics or be notified about certain events that occur in your crawling process.
+Scrapy provides the tools for periodically generate and process data from a :ref:`topics-stats`.
+
+Some use cases can be:
+
+- Monitor and graph scraping data over time (memory usage, http codes, errors, scraped items, etc).
+- Get notified when a crawl job has finished or alerted when something happens.
+- Use your own stat values to monitor custom data or events.
+
+You can for instance generate and export your stats and create some time-based graphs using a 3rd party
+tool like `Kibana`_.
+
+
+To achieve this is you must configure a :ref:`topics-stats-periodicstatscollector` with a list of:
+
+- :ref:`topics-stats-valueobservers`: To select and configure the filtered periodic stats.
+- :ref:`topics-stats-statspipelines` To process the generated stats.
+
+.. _topics-stats-periodicstatscollector:
+
+Periodic Stats Collector
+========================
+
+:class:`PeriodicStatsCollector` is a special :class:`MemoryStatsCollector` class, it has exactly the same funcionality
+but can be configured to periodically generate and process stats.
+
+All configuration is done via settings. To use it, simply change the :setting:`STATS_CLASS` setting like this::
+
+    STATS_CLASS = 'scrapy.contrib.periodicstats.PeriodicStatsCollector'
+
+
+Stat values can be filtered defining :ref:`topics-stats-valueobservers`, but sometimes you want all the available
+stats. You can get this with the :setting:`PERIODIC_STATS_EXPORT_ALL` setting::
+
+    PERIODIC_STATS_EXPORT_ALL = True
+
+And to define the interval for generating all the stats you must use
+the :setting:`PERIODIC_STATS_EXPORT_ALL_INTERVAL` setting::
+
+    PERIODIC_STATS_EXPORT_ALL_INTERVAL = 30  # time in seconds
+
+
+You can always deactivate the periodic stats generation with the :setting:`PERIODIC_STATS_ENABLED` setting
+(:class:`MemoryStatsCollector` functionality will still remain the same)::
+
+    PERIODIC_STATS_ENABLED = False
+
+
+
+.. _topics-stats-valueobservers:
+
+Value Observers
+===============
+
+:class:`PeriodicStatsObserver` class allows to select data from the available :ref:`topics-stats` and define when it
+should be generated.
+
+They can be defined using the :setting:`PERIODIC_STATS_OBSERVERS` setting::
+
+    from scrapy.contrib import periodicstats as stats
+
+    PERIODIC_STATS_OBSERVERS = [
+        stats.Observer(key='downloader/response_count'),
```

Comment:
I really like creating actual parametrized Python objects for the configuration, but for all other Scrapy settings we used strings with class names. The problem with objects is that you need to import scrapy from the settings.py, and this may e.g. cause recursive imports. I'm not sure how big is the problem.